zig-dxread/src/models/dx_data.zig

```diff
@@ -0,0 +1,47 @@
+const std = @import("std");
+const fs = std.fs;
+const constants = @import("../constants.zig");
+const logger = @import("../logger.zig").logger;
+const DXHeader = @import("dx_header.zig").DXHeader;
+const DXRecord = @import("dx_record.zig").DXRecord;
+
+/// DX Data
+/// See section 2.3.3 [here](https://isccp.giss.nasa.gov/pub/documents/d-doc.pdf)
+/// for a full description of this data format.
+pub const DXData = struct {
+    /// Header
+    header: DXHeader,
+    /// List of records within the dx data file.
+    records: std.ArrayList(DXRecord),
+
+    pub fn readFromFile(filename: []const u8, allocator: *std.mem.Allocator) !DXData {
+        const real_filename = try std.fs.realpathAlloc(allocator.*, filename);
+        defer allocator.free(real_filename);
+        var file = try std.fs.openFileAbsolute(real_filename, .{});
+        defer file.close();
+
+        var buf_reader = std.io.bufferedReader(file.reader()).reader();
+        return DXData.readFromReader(buf_reader, allocator);
+    }
+
+    pub fn readFromReader(file_reader: constants.FileReader, allocator: *std.mem.Allocator) !DXData {
+        var record_buffer = try file_reader.readBytesNoEof(constants.RECORD_SIZE);
+        var record_reader: constants.RecordReader = std.io.fixedBufferStream(record_buffer[0..]).reader();
+        var header = try DXHeader.fromReader(record_reader);
```

Comment:
passing in fixed size chunks makes it easier to understand what's going on.
It also makes things hard to trace. If you give too many nested methods control of seeking the whole file it can start to get ugly.

I could have just read in a buffer and indexed it but wanted to try creating a new reader abstraction on top of it out of curiosity. However, I feel that it's not so hard to read, and as you point out, doesn't have the optimization one would expect had we passed down one reader. 
I have changed it to buffers from now on. I think finally this is much clearer. Thanks for the feedback.